up42/auth.py

```diff
@@ -70,16 +61,13 @@ def __init__(
         self.workspace_id: Optional[str] = None
         self.token: Optional[str] = None
 
-        local_vars = locals()
-        credentials_filter = [
-            "project_id",
-            "project_api_key",
-            "username",
-            "password",
-        ]
         credentials_dict = {
-            key: value for key, value in local_vars.items() if key in credentials_filter and value is not None
+            "project_id": project_id,
+            "project_api_key": project_api_key,
+            "username": username,
+            "password": password,
         }
+        credentials_dict = {key: value for key, value in credentials_dict.items() if value}
```

Comment:
nitpick: this will probably never be the case, but since we aren't validating, in theory it is possible to set a bool value to any of the key in `credentials_dict`

If any value is set to `False`, `credentials_dict = {key: value for key, value in credentials_dict.items() if value}` --> this logic would fail.

can we bring back `is not None`?

Reply:
That doesn't make sense for the case of credentials. If some users want authentication to fail, completely ignoring the documentation, let them fail